src/lib.rs

```diff
@@ -1,5 +1,8 @@
 //! Headcrab, a modern Rust debugging library.
 
+#[macro_use]
+extern crate rental;
```

Comment:
This crate seems to be archived & not maintained anymore and the author recommends to explore other solutions: https://github.com/jpernst/rental

As it seems, it's only used for safe borrowing of

Reply:
I didn't know it was archived.

Reply:
I have extracted the actual debuginfo parsing into `ParsedDwarf::new` keeping `Dwarf::new` as a thin wrapper that only handles the self-referential part. I have also written some unsafe code as replacement for rental. I am currently waiting on a review over at the `t-lang/wg-unsafe-code-guidelines` stream of the rust-lang zulip.

Reply:
Thank you!

> some unsafe code as replacement for rental

This can be submitted as a separate issue/PR, if it would be easier for you.
I'm going to review the rest of the code now.

Reply:
I got a review, so I pushed it.